extras/rya.manual/src/site/markdown/rya-streams.md

```diff
@@ -0,0 +1,385 @@
+* Fritz croaks
+* Fritz eats flies
+
+Then the following facts are inferred about Fritz:
+* Fritz is a frog
+* Fritz is green
+
+Rya Streams is able to be used to build an inference engine that executes that
+rule set. SPARQL queries are used to generate new Statements from the original 
+set of Statements. For example:
+
+```
+INSERT {
+    ?x <urn:isA> <urn:frog> 
+} WHERE {
+    ?x <urn:makesSound> <urn:croaks> .
+    ?x <urn:eats> <urn:flies> .
+}
+
+INSERT {
+    ?x <urn:hasColor> <urn:green>
+} WHERE {
+    ?x <urn:isA> <urn:frog>
+}
+```
+As long as the first query's results are ingested into the streams application, 
+the second query will infer that ?x is green based on the previous query's
+inferred statement. 
+
+### PCJ Maintenance ###
+
+Pre-Computed Joins are a feature within Rya where a query's results are 
+maintained and used as an index to speed up query evaluation. See the PCJ
+section of the manual for more information about the specifics of how that
+index works.
+
+Rya Streams is able to create the Visibility Binding Sets that need to be loaded
+into the PCJ index. It just needs to see the same Statements that the core Rya
+instance has been loaded with. 
+
+<div id='future-work'/>
+
+## Future Work ##
+Each of the following subsections cover an improvement that could be made to 
+the Rya Streams system. These improvements would push this feature towards a 
+production ready state.
+
+### Historic Statement Processing ###
+As currently implemented, a program needs to write new Statements to the core
+Rya data store as well as the input topic for Rya Streams. Every time a query
+is added to Rya Streams for processing, it will only see newly added Statements.
+We could write a Kafka Connect Rya source that import historic data into the 
+statements topic so that it, too, will be processed.
+
+### PCJ Maintenance ### 
+The Rya Streams system is capable of creating PCJ results over streamed 
+statements, but there isn't anything written to import those results from the
+QueryResult topic into the source Rya's PCJ index. We could use Kakfa Connect
+to write those binidng sets into the index.
+
+### Query Manager Durability ###
+The Query Manager daemon is a single point of failure for this system. If it
+is not running, then queries can not be processed. This problem could be
+solved in a few different ways depending on the size of the deployed system.
+If multiple machines are available, but only a small amount of data/number of
+queries need to be processed, then failing over to a secondary Query Manager
+would work fine. If a cluster solution is being used, then the Query Manager
+could be refactored to run in something like YARN. It then becomes the
+cluster's responsibility to start up processing on a new machine when the
+application dies.
+
+### Query Change Log Reverse Compatibility ###
+We use Java Serialization to write ChangeLogEntry<QueryChange> objects to the 
```

Comment:
Can't view the `<QueryChange>` part.  It's not being rendered. Use either:
`` `ChangeLogEntry<QueryChange>` ``
or:
`ChangeLogEntry&lt;QueryChange&gt;`

Reply:
Done.